source/common/ur_singleton.hpp

```diff
@@ -76,21 +76,25 @@ template <typename singleton_tn, typename key_tn> class singleton_factory_t {
 
   void retain(key_tn key) {
     std::lock_guard<std::mutex> lk(mut);
-    auto iter = map.find(getKey(key));
-    assert(iter != map.end());
-    iter->second.ref_count++;
+    if (auto iter = map.find(getKey(key)); iter != map.end()) {
+      iter->second.ref_count++;
+    } else {
+      __builtin_unreachable();
```

Comment:
there's a function:
https://github.com/oneapi-src/unified-runtime/blob/8b7a99578966eb691a961d9620ea38d235196b2f/source/common/ur_util.hpp#L339

Reply:
No longer using unreachable in favour of `abort()` - see below.